sql/variables/status_variable_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package variables
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func TestStatusVariables(t *testing.T) {
+	defer InitStatusVariables()
+
+	require := require.New(t)
+	ctx := sql.NewEmptyContext()
+	sess := sql.NewBaseSessionWithClientServer("foo", sql.Client{Address: "baz", User: "bar"}, 1)
+
+	sessVal, err := sess.GetStatusVariable(ctx, "Aborted_clients")
+	require.NoError(err)
+	require.Equal(int64(0), sessVal)
+
+	_, globalVal, ok := sql.StatusVariables.GetGlobal("Aborted_clients")
+	require.True(ok)
+	require.Equal(int64(0), globalVal)
+
+	sess.SetStatusVariable(ctx, "Aborted_clients", int64(100))
+
+	sessVal, err = sess.GetStatusVariable(ctx, "Aborted_clients")
+	require.NoError(err)
+	require.Equal(int64(100), sessVal)
+
+	_, globalVal, ok = sql.StatusVariables.GetGlobal("Aborted_clients")
+	require.True(ok)
+	require.Equal(int64(0), globalVal)
```

Comment:
It's worth thinking a little more about the possible scopes for status variables: global, session, or both. 

For example, `Aborted_clients` has global scope, so if we set it through the session, I'd expect the next call to GetGlobal to also return the newly set value. 

It probably makes more sense to only allow Getting/Setting status variables through the session when their scope is session or both. Then when you access through the session, it's clear that you're getting/setting a status var at a session scope, and when you access through the global API, it's clear that you're getting/setting at a global scope.